src/store/reducer.js

```diff
@@ -29,10 +30,16 @@ function reducer(state = initialState, { type, payload }) {
     const key = `${productId}_${poolId}`;
     const newPoolProduct = { productId, poolId, ...poolProduct };
     const poolProducts = { ...state.poolProducts, [key]: newPoolProduct };
-    const previousIds = state.productPoolIds[productId] || [];
-    const newIds = [...new Set([...previousIds, poolId])];
-    const productPoolIds = { ...state.productPoolIds, [productId]: newIds };
-    return { ...state, poolProducts, productPoolIds };
+
+    const previousPoolIds = state.productPoolIds[productId] || [];
+    const newPoolIds = [...new Set([...previousPoolIds, poolId])];
+    const productPoolIds = { ...state.productPoolIds, [productId]: newPoolIds };
+
+    const previousProductIds = state.poolProductIds[poolId] || [];
+    const newProductIds = [...new Set([...previousProductIds, productId])];
+    const poolProductIds = { ...state.poolProductIds, [poolId]: newProductIds };
+
+    return { ...state, poolProducts, productPoolIds, poolProductIds };
```

Comment:
As discussed: remove `productPoolIds` and `poolProductIds` to remove redundancy and create selectors instead